### ISSUE TITLE
warning in strict-mode: "Unsafe lifecycle methods were found within a strict-mode tree"

### ISSUE THREAD

**Issue by oahehc** [styled-components/styled-components#1864]
when I try [strict-mode](https://reactjs.org/docs/strict-mode.html), all styled-components show this warning,
getDerivedStateFromProps is new API so might no be implemented yet,
how about 'componentWillMount', should this be updated to 'componentDidMount'?
why or why not?

```
Warning: Unsafe lifecycle methods were found within a strict-mode tree:
componentWillMount: Please update the following components to use componentDidMount instead
componentWillReceiveProps: Please update the following components to use static getDerivedStateFromProps instead
```

## System:
 - OS: macOS High Sierra 10.13.5
 - CPU: x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
 - Memory: 86.42 MB / 16.00 GB
 - Shell: 5.3 - /bin/zsh
## Binaries:
 - Node: 8.9.1 - ~/.nvm/versions/node/v8.9.1/bin/node
 - Yarn: 1.3.2 - /usr/local/bin/yarn
 - npm: 6.2.0 - ~/.nvm/versions/node/v8.9.1/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - babel-plugin-styled-components: 1.1.4 => 1.1.4
 - styled-components: 3.2.5 => 3.2.5

## Steps to reproduce
```
const Test = styled.div``;
<React.StrictMode>
   <Test />
</React.StrictMode>
```

## Expected Behavior
no warning message

## Actual Behavior
show warning as above

**Comment by probablyup** [comment]
Yup we're not strict-mode compatible yet. This is [currently slated for 4.0](https://github.com/styled-components/styled-components/issues/1694).
